Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -70,42 +79,21 @@ struct ReportProjectFormView: View {
       .onAppear {
         focusField = .details
         viewModel.inputs.viewDidLoad()
-        viewModel.projectID.send(self.projectID)
-        viewModel.projectFlaggingKind.send(self.projectFlaggingKind)
-      }
-      .onChange(of: details) { detailsText in
-        viewModel.detailsText.send(detailsText)
-      }
-      .onChange(of: saveTriggered) { triggered in
-        focusField = nil
-        showLoading = triggered
-        viewModel.saveTriggered.send(triggered)
       }
-      .onChange(of: bannerMessage) { newValue in
+      .onReceive(viewModel.$bannerMessage) { newValue in
+        showLoading = false
```

Comment:
It might be cleaner to move the banner logic into the ViewModel, so that the submit, banner display, timeout, and dismiss all happen in the same place. However, that was going to be a bigger piece of refactoring, so I left this as-is.

Reply:
Yeah I think that'd be good too